src/components/input/SelectNext.tsx

```diff
@@ -60,7 +60,35 @@ function SelectOption<T>({
   }
 
   const { selectValue, value: currentValue } = selectContext;
-  const selected = !disabled && currentValue === value;
+  const valueIsArray = Array.isArray(currentValue);
+
+  const selected =
+    !disabled &&
+    ((valueIsArray && currentValue.includes(value)) || currentValue === value);
+
+  const selectOrToggle = useCallback(() => {
+    // In single-select, just set current value
+    if (!valueIsArray) {
+      selectValue(value);
+      return;
+    }
+
+    // In multi-select, clear selection for nullish values
+    if (!value) {
+      selectValue([]);
+      return;
+    }
```

Comment:
This is a bit opinionated and not entirely obvious behavior.

Perhaps a better option would be to allow `<SelectOption />` to provide its own `onChange`, in which case it gets invoked instead of the default `selectValue(...)` logic.

Reply:
I suggested above that it might help to avoid errors if the component just threw an error if in multi-select mode and a non-array value is passed.